### PR TITLE
Move WS of project administration to api/projects

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/UpdateKeyAction.java [hunk 1]
```diff
@@ -22,88 +22,24 @@
...
-import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_DEPRECATED_NEW_KEY;
-import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_NEW_PROJECT;
-import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_PROJECT;
-import static org.sonarqube.ws.client.component.ComponentsWsParameters.PARAM_PROJECT_ID;
 
 public class UpdateKeyAction implements ComponentsWsAction {
-  private final DbClient dbClient;
-  private final ComponentFinder componentFinder;
-  private final ComponentService componentService;
+  private final org.sonar.server.project.ws.UpdateKeyAction updateKeyAction;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/UpdateKeyAction.java]: MAJOR Rename field "updateKeyAction" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1700)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-web/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -1,4 +1,24 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (julienlancelot) [inline server/sonar-web/src/main/webapp/WEB-INF/web.xml]: We don't add header on XML files

File: it/it-tests/src/test/java/it/component/ProjectsWsTest.java [hunk 3]
```diff
@@ -71,4 +89,37 @@ public void projects_web_service() throws IOException {
       httpclient.getConnectionManager().shutdown();
     }
   }
+
+  @Test
+  public void update_key() {
```
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/component/ProjectsWsTest.java]: I would move this IT to ProjectAdministrationTest

File: it/it-tests/src/test/java/it/component/ProjectsWsTest.java [hunk 4]
```diff
@@ -71,4 +89,37 @@ public void projects_web_service() throws IOException {
...
+    wsClient.projects().updateKey(UpdateKeyWsRequest.builder()
+      .setKey(PROJECT_KEY)
+      .setNewKey(newProjectKey)
+      .build());
+
+    assertThat(wsClient.components().show(new ShowWsRequest().setId(project.getId())).getComponent().getKey()).isEqualTo(newProjectKey);
+  }
+
+  @Test
+  public void bulk_update_key() {
```
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/component/ProjectsWsTest.java]: I would move this IT to ProjectAdministrationTest

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/BulkUpdateKeyAction.java [hunk 5]
```diff
@@ -100,6 +100,7 @@ public void define(WebService.NewController context) {
     action.createParam(PARAM_PROJECT_ID)
       .setDescription("Project or module ID")
       .setDeprecatedKey("id", "6.4")
+      .setDeprecatedSince("6.4")
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/BulkUpdateKeyAction.java]: Why ?

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/DeleteAction.java [hunk 6]
```diff
@@ -65,6 +65,7 @@ public void define(WebService.NewController context) {
       .createParam(PARAM_PROJECT_ID)
       .setDescription("Project ID")
       .setDeprecatedKey("id", "6.4")
+      .setDeprecatedSince("6.4")
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/DeleteAction.java]: Why ?

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/UpdateKeyAction.java [hunk 7]
```diff
@@ -74,6 +74,7 @@ public void define(WebService.NewController context) {
     action.createParam(PARAM_PROJECT_ID)
       .setDescription("Project or module id")
       .setDeprecatedKey("id", "6.4")
+      .setDeprecatedSince("6.4")
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/UpdateKeyAction.java]: why ?

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ComponentFinder.java#L109](https://github.com/SonarSource/sonarqube/blob/2bd31891d38a8d6364004ac9ae0300cdd6366566/server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java#L109): Define a constant instead of duplicating this literal "projectId" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**